test/view/board_editor/board_editor_test.dart

```diff
@@ -0,0 +1,278 @@
+        tester
+            .widget<BottomBarButton>(
+              find.byKey(const Key('analysis-board-button')),
+            )
+            .onTap,
+        isNull,
+      );
+    });
+
+    testWidgets('Delete pieces via bin button', (tester) async {
+      final boardEditorController = await setupBoardEditor(tester);
+
+      await tester.tap(find.byKey(const Key('delete-button-white')));
+      await tester.pump();
+
+      await tapSquare(tester, 'e2');
+      expect(
+        boardEditorController.state.fen,
+        'rnbqkbnr/pppppppp/8/8/8/8/PPPP1PPP/RNBQKBNR w KQkq - 0 1',
+      );
+
+      // Change back to drag mode -> tapping has no effect anymore
+      await tester.tap(find.byKey(const Key('drag-button-white')));
+      await tester.pump();
+      await tapSquare(tester, 'e3');
+      expect(
+        boardEditorController.state.fen,
+        'rnbqkbnr/pppppppp/8/8/8/8/PPPP1PPP/RNBQKBNR w KQkq - 0 1',
+      );
+
+      // Now remove all of black's pawns
+      await tester.tap(find.byKey(const Key('delete-button-black')));
+      await tester.pump();
+      await panFromTo(tester, 'a7', 'h7');
+
+      expect(
+        boardEditorController.state.fen,
+        'rnbqkbnr/8/8/8/8/8/PPPP1PPP/RNBQKBNR w KQkq - 0 1',
+      );
+    });
+
+    testWidgets('Add pieces via tap and pan', (tester) async {
+      final boardEditorController = await setupBoardEditor(tester);
+
+      await tester.tap(find.byKey(const Key('piece-button-white-queen')));
+      await panFromTo(tester, 'a1', 'a8');
+      await tester.tap(find.byKey(const Key('piece-button-black-rook')));
+      await tapSquare(tester, 'h1');
+      await tapSquare(tester, 'h3');
+
+      expect(
+        boardEditorController.state.fen,
+        'Qnbqkbnr/Qppppppp/Q7/Q7/Q7/Q6r/QPPPPPPP/QNBQKBNr w k - 0 1',
+      );
+    });
+
+    testWidgets('Drag pieces onto the board', (tester) async {
+      final boardEditorController = await setupBoardEditor(tester);
+
+      // Start by pressing bin button, dragging a piece should override this
+      await tester.tap(find.byKey(const Key('delete-button-black')));
+      await tester.pump();
+
+      final pieceButtonOffset =
+          tester.getCenter(find.byKey(const Key('piece-button-white-pawn')));
+      await tester.dragFrom(
+        pieceButtonOffset,
+        tester.getCenter(find.byKey(const Key('d3-empty'))) - pieceButtonOffset,
+      );
+      await tester.dragFrom(
+        pieceButtonOffset,
+        tester.getCenter(find.byKey(const Key('d1-whitequeen'))) -
+            pieceButtonOffset,
+      );
+
+      expect(
+        boardEditorController.state.editorPointerMode,
+        EditorPointerMode.drag,
+      );
+
+      expect(
+        boardEditorController.state.fen,
+        'rnbqkbnr/pppppppp/8/8/8/3P4/PPPPPPPP/RNBPKBNR w KQkq - 0 1',
+      );
+    });
+  });
+}
+
+Future<BoardEditorController> setupBoardEditor(WidgetTester tester) async {
+  final boardEditorController = BoardEditorController();
```

Comment:
It is not ideal to construct directly a notifier instance. Testing providers should be done according to Riverpod's doc:
https://riverpod.dev/docs/essentials/testing

Creating/reading providers should always go through a `Container`.